### DISCUSSION TITLE
vue2 用了render函数报 React is not defined

### DISCUSSION THREAD

**Discussion by xiemeng** [web-infra-dev/rsbuild#2328] [Q&A]
![1715655619945](https://github.com/web-infra-dev/rsbuild/assets/24853184/ed261b34-f34c-42a9-ae83-de6c8492ec26)
![1715655714915](https://github.com/web-infra-dev/rsbuild/assets/24853184/82ad3254-1440-420b-adbf-681965b93de8)
不使用render函数就正常
`import { pluginVue2 } from '@rsbuild/plugin-vue2'
import { defineConfig, loadEnv } from '@rsbuild/core'
import { pluginBabel } from '@rsbuild/plugin-babel'
import { pluginVue2Jsx } from '@rsbuild/plugin-vue2-jsx'
const { publicVars } = loadEnv({ prefixes: ['VUE_APP_'] })
const path = require('path')
const isProd = process.env.NODE_ENV === 'production'
const assetsCDN = {
  externals: {
    // vue: 'Vue',
    // 'vue-router': 'VueRouter',
    // vuex: 'Vuex',
    // axios: 'axios'
  },
  css: [],
  // https://unpkg.com/browse/vue@2.6.10/
  js: [
    '//cdn.jsdelivr.net/npm/vue@2.6.10/dist/vue.min.js',
    '//cdn.jsdelivr.net/npm/vue-router@3.1.3/dist/vue-router.min.js',
    '//cdn.jsdelivr.net/npm/vuex@3.1.1/dist/vuex.min.js',
    '//cdn.jsdelivr.net/npm/axios@0.19.0/dist/axios.min.js',
  ],
}
function resolve(dir) {
  return path.join(__dirname, dir)
}
export default defineConfig({
  plugins: [
    pluginVue2(),
    pluginBabel({
      include: /\.(?:jsx|tsx)$/,
      exclude: /[\\/]node_modules[\\/]/,
    }),
    pluginVue2Jsx(),
  ],
  source: {
    // 指定入口文件
    entry: {
      index: './src/main.ts',
    },
    define: publicVars,
  },
  html: {
    template: './public/index.html',
  },
  tools: {
    cssLoader: {},
    less: {
      // additionalData: `@import "${resolve(__dirname, 'src')}/style/var.less";`,
      implementation: require('less'),
      lessOptions: {
        modifyVars: {
          // less vars，customize ant design theme

          // 'primary-color': '#F5222D',
          // 'link-color': '#F5222D',
          'border-radius-base': '2px',
        },
        // DO NOT REMOVE THIS LINE
        javascriptEnabled: true,
      },
    },
    rspack: {
      resolve: { extensions: ['.ts', '.tsx', '.js', '.json'] },
      mode: !isProd ? 'development' : 'production',
      plugins: [
        // new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
      ],
      // if prod, add externals
      externals: isProd ? assetsCDN.externals : {},
    },
    bundlerChain: (config) => {
      config.resolve.alias.set('@$', resolve('src'))
      const svgRule = config.module.rule('svg')
      svgRule.uses.clear()
      svgRule
        .oneOf('inline')
        .resourceQuery(/inline/)
        .use('vue-svg-icon-loader')
        .loader('vue-svg-icon-loader')
        .end()
        .end()
        .oneOf('external')
        .use('file-loader')
        .loader('file-loader')
        .options({
          name: 'assets/[name].[hash:8].[ext]',
        })
    },
  },
  server: {
    port: 8000,
    proxy: {
      '/api': {
        target: 'https://',
        ws: true,
        changeOrigin: true,
        logLevel: 'debug',
        pathRewrite: {
          '^/api': '',
        },
      },
    },
  },
})
`

**Comment by chenjiahan** [comment]
Can you try to add `lang="jsx"` or `lang="tsx"`  in your Vue SFC file?

```diff
- <script>
+ <script lang="jsx">
```
